homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,229 @@
+"""Config flow for LastFm."""
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import Any
+
+from pylast import LastFMNetwork, User, WSError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_MAIN_USER, CONF_USERS, DOMAIN, LOGGER
+
+PLACEHOLDERS = {"api_account_url": "https://www.last.fm/api/account/create"}
+
+INITIAL_CONFIG_SCHEMA: vol.Schema = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+        vol.Required(CONF_MAIN_USER): str,
+    }
+)
+
+
+class LastFmFlowHandler:
+    """Base handler for LastFM flows."""
+
+    def __init__(self, data: dict[str, Any]) -> None:
+        """Initialize LastFM flow."""
+        self.data = data
+
+    def _get_lastfm_user(self, username: str) -> User:
+        return LastFMNetwork(api_key=self.data[CONF_API_KEY]).get_user(username)
+
+    def _validate_lastfm_user(self, user: User) -> dict[str, str] | None:
```

Comment:
You can put both these functions outside of the class and skip this base class all together